### ISSUE TITLE
Enable torch.nn.modules.* typechecks during CI

### ISSUE THREAD

**Issue by guilhermeleobas** [pytorch/pytorch#46012]
`TestTypeHints.test_run_mypy` should pass with `[mypy-torch.nn.modules.(loss/padding/pooling/rnn)]` deleted from mypy.ini

cc @ezyang @malfet @rgommers @xuzhao9 @gramster
